ykrt/try_passes.py

```diff
@@ -29,19 +27,28 @@ def __str__(self):
         link_time = ",".join([ str(p) for p in self.link_time ])
         return f"PipelineConfig(pre_link=[{pre_link}], link_time=[{link_time}])"
 
-# Define the pipeline in advance.
-PIPELINE_CONFIG_MAP = {
-    'lto': lambda ok_passes, try_passes: PipelineConfig([], ok_passes + try_passes),
-    'prelink': lambda ok_passes, try_passes: PipelineConfig(ok_passes + try_passes, []),
-    'O2': lambda ok_passes, try_passes: PipelineConfig(ok_passes + try_passes, ok_passes + try_passes)
-}
-
-OPT_CMD_MAP = {
-    'lto': "opt -passes='lto<O2>' -print-pipeline-passes < /dev/null 2>/dev/null | tr ',' '\n'",
-    'prelink': "opt -passes='lto-pre-link<O2>' -print-pipeline-passes < /dev/null 2>/dev/null | tr ',' '\n'",
-    'O2': "opt -O2 -print-pipeline-passes < /dev/null 2>/dev/null | tr ',' '\n'"
-}
-# Command to run for flagType i.e -lto, -prelink, -O2`.
+def get_pipeline_config(flagType, ok_passes, try_passes):
+    """Returns the appropriate pipeline configuration based on the flagType."""
+    
+    if flagType == 'lto':
```

Comment:
If you use a bool for the flag type then you don't need to have an else case.

Reply:
I don't understand, currently we have two options `lto` or `prelink` if user pass any other flag we raise an error. Do we instead want to have a default flagType set if the user pass anything other than these two options?

Reply:
I'd replace the string `flagType` with a bool `pre_link`. There are only two possibilities, so a bool covers both nicely.

Reply:
Fixed here: 9e17fdb52b14807eefd50bfe0335386fecfe9121